libqtile/backend/x11/window.py

```diff
@@ -1077,6 +1077,20 @@ def __init__(self, win, qtile, screen,
         self.place(self.x, self.y, self.width, self.height, 0, 0)
         self.update_strut()
 
+        # Grab button 1 to focus upon click
+        for amask in self.qtile.core._auto_modmasks():
+            self.qtile.core.conn.conn.core.GrabButton(
+                True,
+                self.window.wid,
+                EventMask.ButtonPress,
+                xcffib.xproto.GrabMode.Sync,
+                xcffib.xproto.GrabMode.Async,
+                xcffib.xproto.Atom._None,
+                xcffib.xproto.Atom._None,
+                1,
+                amask,
+            )
+
```

Comment:
Why not move this into `_Window.__init__` rather than copy paste it?

Reply:
Because we don't want it on `Internal` windows

Reply:
Wouldn't it make more sense if internal windows worked the same way than the other windows? Do you know why it's not the case at the moment?

Reply:
They mostly do, it's just that this new button grabbing is _only_ required so that we can focus the window and pass the event down to the window cleanly, but we don't want to focus internal windows so don't need this implemented on those windows. The cost to doing so is some hackery involving the xembed stuff around the Systray, which seems unnecessary given that this solution works anyway.